packages/format-library/src/link/index.js

```diff
@@ -38,10 +38,38 @@ function Edit( {
 	onFocus,
 	contentRef,
 } ) {
-	const [ addingLink, setAddingLink ] = useState( false );
+	const [ editingLink, setEditingLink ] = useState( false );
+	const [ creatingLink, setCreatingLink ] = useState( false );
+
 	// We only need to store the button element that opened the popover. We can ignore the other states, as they will be handled by the onFocus prop to return to the rich text field.
 	const [ openedBy, setOpenedBy ] = useState( null );
 
+	const [ autoFocus, setAutoFocus ] = useState( true );
+
+	function setIsEditingLink( _val, { shouldAutoFocus = true } = {} ) {
+		setEditingLink( _val );
+		setAutoFocus( shouldAutoFocus );
+	}
+
+	function setIsCreatingLink( _val ) {
+		// Don't add a new link if there is already an active link.
+		// The two states are mutually exclusive.
+		if ( _val === true && isActive ) {
+			return;
+		}
+		setCreatingLink( _val );
+	}
+
+	useEffect( () => {
+		// When the link becomes inactive (i.e. isActive is false), reset the editingLink state
+		// and the isAddingNewLink state. This means that if the Link UI is displayed and the link
```

Comment:
is there a difference why isAddingNewLink but only editingLink (not isEditingLink)?

Reply:
No - its a typo I will correct.